cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/DatabaseBrowser/index.js

```diff
@@ -35,29 +35,28 @@ require('./DatabaseBrowser.scss');
 const PREFIX = `features.DataPrep.DataPrepBrowser.DatabaseBrowser`;
 
 export default class DatabaseBrowser extends Component {
-  constructor(props) {
-    super(props);
-    let store = DataPrepBrowserStore.getState();
-    this.state = {
-      info: store.database.info,
-      connectionId: store.database.connectionId,
-      connectionName: '',
-      tables: [],
-      loading: true,
-      search: '',
-      searchFocus: true,
-      error: null
-    };
+  static propTypes = {
+    toggle: PropTypes.func,
+    onWorkspaceCreate: PropTypes.func
+  };
 
-    this.eventEmitter = ee(ee);
-    this.handleSearch = this.handleSearch.bind(this);
-    this.prepTable = this.prepTable.bind(this);
-    this.eventBasedFetchTable = this.eventBasedFetchTable.bind(this);
+  store = DataPrepBrowserStore.getState();
```

Comment:
this store is only being used once for the initialization of the `state` object. It's better to just do the getState inline for the initialization and not clutter the `this` context. I believe this object won't get garbage collected until the class instance had been decommissioned